### PR TITLE
[WA-30] favorite list with specific weather

### PR DESCRIPTION
**Pull request by uselesswhale1**
- Fav-cities use requests to get current information about weather
- Implemented storage services to favorite-list
- Added fav-city adding on icon heart

### CODE REVIEW

File: src/app/home/components/favorite-list/favorite-list.component.ts [hunk 1]
```diff
@@ -16,19 +15,17 @@ import { NotificationType } from 'src/app/shared/enums/notification.enum';
...
-      this.favoriteService.getCurrentWeatherForecast(city, 1)
-      .subscribe((city) => {
-        this.favorites.push(city);
-      });
-    });
-    console.log(this.favorites);
-    if(this.favoriteService.checkAmountOfFavorites()){
-      this.notificationService.show( NotificationType.Error ,'Your can add only 10 cities to your list of favorites')
+    this.favorites = this.favoriteService.getFavorites();
+    if(this.favoriteStateService.moreCitiesCanBeAdded()){
```
Comment (dmitry-chirva) [inline src/app/home/components/favorite-list/favorite-list.component.ts:27]: rename to hasMaxCities

File: src/app/shared/components/favorite-toggle/favorite-toggle.component.ts [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-  constructor(private notificationService: NotificationService) {}
+  constructor(private notificationService: NotificationService, private favoriteStateService: FavoriteStateService) {}
   @Input() isFavorite: boolean | undefined;
   @Output() onChange = new EventEmitter<boolean | undefined>();
 
   onToggle() {
     this.isFavorite = !this.isFavorite;
     this.onChange.emit(this.isFavorite);
-
+    this.favoriteStateService.addFavoriteCity('Toronto');
```
Comment (dmitry-chirva) [inline src/app/shared/components/favorite-toggle/favorite-toggle.component.ts:18]: if it is for test remove it please

File: src/app/home/components/favorite-list/favorite-list.component.ts (hunk 1 above)
Comment (uselesswhale1) [reply src/app/home/components/favorite-list/favorite-list.component.ts:27]: > rename to hasMaxCities

renamed

File: src/app/shared/components/favorite-toggle/favorite-toggle.component.ts (hunk 2 above)
Comment (uselesswhale1) [reply src/app/shared/components/favorite-toggle/favorite-toggle.component.ts:18]: > 

removed

File: src/core/favorites-state/favorite-state.service.ts [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
+  readonly MAX_AMOUNT_OF_FAVS = 10;
   constructor(private storage: StorageService) {}
 
   getFavoriteCities(): string[] {
     return this.storage.getItem(this.FAVORITES_KEY);
   }
 
   addFavoriteCity(city: string): void {
-    this.storage.addItem(this.FAVORITES_KEY, city);
+    if(this.isFavoriteCity(city)){
```
Comment (dmitry-chirva) [inline src/core/favorites-state/favorite-state.service.ts:17]: if(this.isFavoriteCity(city) && this.hasMaxCities()) {
  this.storage.addItem(this.FAVORITES_KEY, city);
}

File: src/core/favorites-state/favorite-state.service.ts [hunk 4]
```diff
@@ -25,4 +28,8 @@ export class FavoriteStateService {
   isFavoriteCity(city: string): boolean {
     return this.storage.getItem(this.FAVORITES_KEY).indexOf(city) === -1;
   }
+
+  moreCitiesCanBeAdded(): boolean{
+    return (this.storage.getItem(this.FAVORITES_KEY).length > this.MAX_AMOUNT_OF_FAVS) ? false : true;
```
Comment (dmitry-chirva) [inline src/core/favorites-state/favorite-state.service.ts:33]: just this.storage.getItem(this.FAVORITES_KEY).length > this.MAX_AMOUNT_OF_FAVS

Comment (uselesswhale1) [reply src/core/favorites-state/favorite-state.service.ts:33]: fixed method

**Pull request merged**